hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/utils/TransactionUtils.java

```diff
@@ -72,7 +72,8 @@ public static Option<HoodieCommitMetadata> resolveWriteConflictIfAny(
       boolean timelineRefreshedWithinTransaction,
       Set<String> pendingInstants) throws HoodieWriteConflictException {
     WriteOperationType operationType = thisCommitMetadata.map(HoodieCommitMetadata::getOperationType).orElse(null);
-    if (config.needResolveWriteConflict(operationType)) {
+    if (config.needResolveWriteConflict(operationType, table.isMetadataTable(),
+        config.isStreamingWritesToMetadataEnabled(table.getMetaClient().getTableConfig().getTableVersion()))) {
```

Comment:
Instead of adding these new arguments, is there a way to just compute or fetch these values in `needResolveWriteConflict`?